// react-genie-styled-components/dist/react-genie-styled-components.esm.js
import bounceInLeft from 'react-animations/lib/bounce-in-left';
import bounceInRight from 'react-animations/lib/bounce-in-right';
import fadeIn from 'react-animations/lib/fade-in';
import fadeInLeft from 'react-animations/lib/fade-in-left';
import fadeInRight from 'react-animations/lib/fade-in-right';
import slideInLeft from 'react-animations/lib/slide-in-left';
import slideInRight from 'react-animations/lib/slide-in-right';
import { createGlobalStyle, keyframes } from 'styled-components';

var translate3d = function translate3d(a, b, c) {
  return "translate3d(" + a + ", " + b + ", " + c + ")";
};

var fadeInUp = {
  from: {
    opacity: 0,
    transform: /*#__PURE__*/translate3d(0, '50px', 0)
  },
  to: {
    opacity: 1,
    transform: 'none'
  }
};

function _taggedTemplateLiteralLoose(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }

  strings.raw = raw;
  return strings;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["", ""]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["", ""]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["", ""]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["", ""]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["", ""]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["", ""]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["", ""]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["", ""]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n\n  .", " {\n    animation: ", " 500ms;\n  } \n  \n  .", " {\n    animation: ", " 500ms;\n  }\n  \n  .", " {\n    animation: ", " 500ms;\n  }\n  \n  .", " {\n    animation: ", " 500ms;\n  }\n\n  .", " {\n    animation: ", " 500ms;\n  }\n\n  .", " {\n    animation: ", " 500ms;\n  }\n\n  .", " {\n    animation: ", " 1250ms;\n  }\n\n  .", " {\n    animation: ", " 1250ms;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Animation;

(function (Animation) {
  Animation["FadeInUp"] = "fade-in-up";
  Animation["FadeIn"] = "fade-in";
  Animation["SlideInLeft"] = "slide-in-left";
  Animation["SlideInRight"] = "slide-in-right";
  Animation["FadeInLeft"] = "fade-in-left";
  Animation["FadeInRight"] = "fade-in-right";
  Animation["BounceInLeft"] = "bounce-in-left";
  Animation["BounceInRight"] = "bounce-in-right";
})(Animation || (Animation = {}));

var ReactGenieAnimations = /*#__PURE__*/createGlobalStyle( /*#__PURE__*/_templateObject(), Animation.FadeInUp, /*#__PURE__*/keyframes( /*#__PURE__*/_templateObject2(), fadeInUp), Animation.FadeIn, /*#__PURE__*/keyframes( /*#__PURE__*/_templateObject3(), fadeIn), Animation.SlideInLeft, /*#__PURE__*/keyframes( /*#__PURE__*/_templateObject4(), slideInLeft), Animation.SlideInRight, /*#__PURE__*/keyframes( /*#__PURE__*/_templateObject5(), slideInRight), Animation.FadeInLeft, /*#__PURE__*/keyframes( /*#__PURE__*/_templateObject6(), fadeInLeft), Animation.FadeInRight, /*#__PURE__*/keyframes( /*#__PURE__*/_templateObject7(), fadeInRight), Animation.BounceInLeft, /*#__PURE__*/keyframes( /*#__PURE__*/_templateObject8(), bounceInLeft), Animation.BounceInRight, /*#__PURE__*/keyframes( /*#__PURE__*/_templateObject9(), bounceInRight));

export { Animation, ReactGenieAnimations, fadeInUp };
